### PR TITLE
Fix linting issues

### PR DESCRIPTION
**Pull request by gardar**
fixes: #283

### CODE REVIEW

File: meta/main.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 galaxy_info:
   author: Erik-jan Riemers
+  role_name: gitlab_runner
```
Comment (guenhter) [inline meta/main.yml:4]: @riemers Do you think this is fine?

**Comment by guenhter** [comment]
@gardar If you could push a merge request with the first three commits (without "fix role name"), I'd merge that right away. Those three commits are not really isolated so they can be squashed together. 
The commit "fix role name" needs some attention because we don't wanna mess again the Ansible galaxy, this is why I'd separate that single commit.

**Comment by gardar** [comment]
ansible-lint refuses to run without fixing the role name first.
With that being said I'm still working on this, so it's not ready for merge just yet.

**Comment by gardar** [comment]
All issues fixed, are you sure about reverting the [role name](https://ansible.readthedocs.io/projects/lint/rules/role-name/) from `gitlab_runner` to `gitlab-runner`?

**Comment by riemers** [comment]
It should be as it is in ansible galaxy right now. Since that one has all the history.
________________________________
Van: gardar ***@***.***>
Verzonden: Tuesday, January 23, 2024 5:22:32 PM
Aan: riemers/ansible-gitlab-runner ***@***.***>
CC: Erik-jan Riemers ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [riemers/ansible-gitlab-runner] Fix linting issues (PR #312)


All issues fixed, are you sure about reverting the role name<https://ansible.readthedocs.io/projects/lint/rules/role-name/> from gitlab_runner to gitlab-runner?

**Comment by gardar** [comment]
I see, the transition from hyphens to underscore in galaxy was some time ago and I think you can't make new roles with hyphens in the name.
Perhaps it's possible to get an alias in galaxy? So that both `gitlab-runner` and `gitlab_runner` names work?

**Comment by riemers** [comment]
Don’t dare to touch it.. but I can ask if an alias is possible. But if it is that means people can steal other names too. So I doubt it. Going back and forth also doesn’t help ;)
________________________________
Van: gardar ***@***.***>
Verzonden: Tuesday, January 23, 2024 5:40:27 PM
Aan: riemers/ansible-gitlab-runner ***@***.***>
CC: Erik-jan Riemers ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [riemers/ansible-gitlab-runner] Fix linting issues (PR #312)


I see, the transition from hyphens to underscore in galaxy was some time ago and I think you can't make new roles with hyphens in the name.
Perhaps it's possible to get an alias in galaxy? So that both gitlab-runner and gitlab_runner names work?

**Comment by gardar** [comment]
Well since role names are now required to be with underscores and hyphens are only allowed for legacy purposes one would assume an alias would be standard practice.
I don't think anyone is able to steal your role name since it's prefixed with your namespace 😃 

Let me know if you want to check with the galaxy team or I can change the metadata and add the role name linting rule to the ignore list.

**Comment by riemers** [comment]
If it is possible to have both that would be fine too. I know a lot of people used the old role.. but it won’t be end of the world either.
________________________________
Van: gardar ***@***.***>
Verzonden: Thursday, January 25, 2024 2:52:02 PM
Aan: riemers/ansible-gitlab-runner ***@***.***>
CC: Erik-jan Riemers ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [riemers/ansible-gitlab-runner] Fix linting issues (PR #312)


Well since role names are now required to be with underscores and hyphens are only allowed for legacy purposes one would assume an alias would be standard practice.
I don't think anyone is able to steal your role name since it's prefixed with your namespace 😃

Let me know if you want to check with the galaxy team or I can change the metadata and add the role name linting rule to the ignore list.

**Comment by gardar** [comment]
Ok let me know if you get any response from the inquiry, otherwise I'll change the meta.

**Comment by ericsysmin** [comment]
> All issues fixed, are you sure about reverting the [role name](https://ansible.readthedocs.io/projects/lint/rules/role-name/) from `gitlab_runner` to `gitlab-runner`?

In newer Ansible versions gitlab-runner throws errors, using _ in the role name is basically a requirement. Using the role_name value in meta helps address this without changing repository name.

**Comment by riemers** [comment]
Maybe we can do both? I mean if it’s a requirement somewhere on the website and we should follow that pattern but then I loose the history / downloads from it too no?
________________________________
Van: Eric Anderson ***@***.***>
Verzonden: Monday, February 12, 2024 7:56:17 PM
Aan: riemers/ansible-gitlab-runner ***@***.***>
CC: Erik-jan Riemers ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [riemers/ansible-gitlab-runner] Fix linting issues (PR #312)


All issues fixed, are you sure about reverting the role name<https://ansible.readthedocs.io/projects/lint/rules/role-name/> from gitlab_runner to gitlab-runner?

In newer Ansible versions gitlab-runner throws errors, using _ in the role name is basically a requirement. Using the role_name value in meta helps address this without changing repository name.

**Comment by gardar** [comment]
So what's the situation here, have you checked if the role can be renamed or if you can get a name alias?

**Comment by riemers** [comment]
Been I’ll for some time and also busy as you are also ;) still have to do boring taxes and playing helpdesk for the family’s. I’ll need some time in the weekend and take some time for it.
________________________________
Van: gardar ***@***.***>
Verzonden: Tuesday, February 27, 2024 5:02:40 PM
Aan: riemers/ansible-gitlab-runner ***@***.***>
CC: Erik-jan Riemers ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [riemers/ansible-gitlab-runner] Fix linting issues (PR #312)


So what's the situation here, have you checked if the role can be renamed or if you can get a name alias?

**Comment by gardar** [comment]
Any news @riemers ?

**Comment by guenhter** [comment]
@gardar If you would split this PR to have only linting changes and not metadata changes like the role_name, I'll be happy to merge it right away. If you keep it as is, I don't want to touch it because we had troubles with Ansible-Galaxy before and this is too hot for me.

**Comment by guenhter** [comment]
And let me know when you are done, then I'll have another look at it

**Comment by gardar** [comment]
Thanks! Haven't had time to complete it yet, I'll give you a ping once it's ready.

**Comment by gardar** [comment]
Unstale

**Comment by riemers** [comment]
Unstale i suppose ;p

File: requirements.yml
Comment (Stunkymonkey) [inline requirements.yml:1]: maybe we should close #342. the linting would already "fix" it. but not with optional dependencies.

**Comment by riemers** [comment]
See #344

**Pull request closed without merging**